github/util.py

```diff
@@ -53,12 +53,15 @@ def __init__(
         github_config: GithubConfig,
         repo_owner: str,
         repo_name: str,
-        branch: str,
+        branch: str = None,
```

Comment:
hmm.. starting to think we require an additional type `GitHubRepo` (which GHRB might extend). I don't think it makes much sense to have a type that explicitly states it be a container type whose instances represent "a specific branch of a given GitHub repository" and then allow certain instances to _not_ specify a branch.

Reply:
it might also be an option to rename it to just `GitHubRepo` and then make the branch name optional.